docs/developer/decision-records/2022-02-03-integration-testing/README.md

```diff
@@ -0,0 +1,301 @@
+if required in the future.
+
+## Spikes
+
+We have performed technical spikes on [sample 04.0-file-transfer](../../../samples/04.0-file-transfer/README.md), that runs two EDC connectors, a Consumer and a Provider. [Spike code is in a forked repository](https://github.com/agera-edc/DataSpaceConnector/pull/3/files). The test requires three components:
+
+- Consumer EDC connector
+- Provider EDC connector
+- HTTP client code to interact with the Consumer provider API
+
+We have written code for the following three options:
+
+- **Docker-compose,** which works but provides an inconvenient inner loop.
+- **Class Loader with Gradle Classpath**, which uses Gradle to determine the effective class path for each runtime, and is very efficient. It is therefore the base for the recommendation above.
+- **Class Loader with Shadow JAR**, a variant of the above which uses the Shadow JAR, and provides a less efficient inner loop.
+
+### Docker-compose
+
+In this setup, we run both connectors in Docker containers. Once they are up, we run HTTP client code in a JUnit test running on the host.
+
+```yaml
+# docker-compose.yaml
+services:
+  sample04-connector-provider:
+    build:
+      context: .
+      target: sample04-connector-provider
+    ports:
+      - "8181:8181"
+    volumes:
+      - /tmp/provider:/tmp/provider
+      - /tmp/consumer:/tmp/consumer
+
+  sample04-connector-consumer:
+    build:
+      context: .
+      target: sample04-connector-consumer
+    ports:
+      - "9191:9191"
+    volumes:
+      - /tmp/provider:/tmp/provider
+      - /tmp/consumer:/tmp/consumer
+
+```
+
+```shell
+$ (cd samples/04.0-file-transfer && docker-compose up -d)
+
+[...]
+Successfully built 26b639e8f852
+Successfully tagged 040-file-transfer_sample04-connector-consumer:latest
+
+[+] Running 3/3
+ ⠿ Network 040-file-transfer_default      Created                                                                                                                                0.0s
+ ⠿ Container sample04-connector-provider  Started                                                                                                                                0.7s
+ ⠿ Container sample04-connector-consumer  Started                                                                                                                                0.6s
+```
+
+```shell
+RUN_INTEGRATION_TEST=true EDC_PROVIDER_CONNECTOR_HOST=http://sample04-connector-provider:8181 time ./gradlew cleanTest :samples:04.0-file-transfer:integration-tests:test --tests org.eclipse.dataspaceconnector.samples.FileTransferSystemTest
+```
+
+This setup is stable and straightforward, but the inner loop is not very efficient. Debugging a remote process is possible buts adds complexity. For faster debugging one approach could be is to run connectors in debug mode within IDE and later use the same code to update docker build. 
```

Comment:
The IDE debug option is a bit too hidden in this document IMO. It enables a very efficient inner loop too, and default run configurations can be committed to the repository.

Reply:
Updated the doc to make it more clearer and also provided more context.